diff --git a/node_modules/@startupjs/is-expression/index.js b/node_modules/@startupjs/is-expression/index.js
index b887946..a0cdffa 100644
--- a/node_modules/@startupjs/is-expression/index.js
+++ b/node_modules/@startupjs/is-expression/index.js
@@ -1,6 +1,6 @@
 'use strict';
 
-var acorn = require('acorn');
+var babelParser = require('@babel/parser');
 var objectAssign = require('object-assign');
 
 module.exports = isExpression;
@@ -16,24 +16,11 @@ function isExpression(src, options) {
   options = objectAssign({}, DEFAULT_OPTIONS, options);
 
   try {
-    var parser = new acorn.Parser(options, src, 0);
-
-    if (options.strict) {
-      parser.strict = true;
-    }
-
-    if (!options.lineComment) {
-      parser.skipLineComment = function (startSkip) {
-        this.raise(this.pos, 'Line comments not allowed in an expression');
-      };
-    }
-
-    parser.nextToken();
-    parser.parseExpression();
-
-    if (parser.type !== acorn.tokTypes.eof) {
-      parser.unexpected();
-    }
+    babelParser.parseExpression(src, {
+      sourceType: 'module',
+      strictMode: !!options.strict,
+      plugins: ['typescript'],
+    });
   } catch (ex) {
     if (!options.throw) {
       return false;
diff --git a/node_modules/@startupjs/is-expression/package.json b/node_modules/@startupjs/is-expression/package.json
index c9ab2cf..00b550c 100644
--- a/node_modules/@startupjs/is-expression/package.json
+++ b/node_modules/@startupjs/is-expression/package.json
@@ -13,6 +13,7 @@
     "expression"
   ],
   "dependencies": {
+    "@babel/parser": "^7.0.0",
     "acorn": "^8.8.2",
     "object-assign": "^4.1.1"
   },
